graphics/fonts/bdf.cpp

```diff
@@ -787,59 +800,150 @@ BdfFont *BdfFont::scaleFont(const BdfFont *src, int newSize) {
 		const BdfBoundingBox &box = data.boxes ? data.boxes[i] : data.defaultBox;
 		const BdfBoundingBox &srcBox = data.boxes ? src->_data.boxes[i] : src->_data.defaultBox;
 
-		if (src->_data.bitmaps[i]) {
-			const int bytes = ((box.width + 7) / 8) * box.height; // Dimensions have been already corrected
-			bitmaps[i] = new byte[bytes];
+		// int hs = srcBox.height;
+		int h = box.height;
+		int ws = srcBox.width;
+		int w = box.width;
+		int grayLevel = h * w / 3;
+		int dstPitch = (w + 7) / 8 ;
 
-			int srcPitch = (srcBox.width + 7) / 8;
-			int dstPitch = (box.width + 7) / 8;
+#if DRAWDEBUG
+		int ccc = 'a';
+		dododo11 = i == ccc;
+#endif
+		srcSurf.fillRect(Common::Rect(srcSurf.w, srcSurf.h), 0);
+		src->drawChar(&srcSurf, i + src->_data.firstCharacter, box.xOffset, 0, 1);
+		memset(dstGray, 0, dstGraySize * sizeof(int));
+		magnifyGray(&srcSurf, dstGray, ws, src->_data.height, scale);
 
-			byte *ptr = bitmaps[i];
+		if (src->_data.bitmaps[i]) {
+			int *grayPtr = dstGray;
+			for (int y = 0; y < h; y++) {
+				byte *dst = (byte *)srcSurf.getBasePtr(0, y);
 
-			for (int y = 0; y < box.height; y++) {
-				const byte *srcd = (const byte *)&src->_data.bitmaps[i][((int)((float)y / scale)) * srcPitch];
+				for (int x = 0; x < w; x++, grayPtr++, dst++) {
+#if DRAWDEBUG
+					if (i == ccc) {
+						if (*grayPtr)
+							debugN(1, "%3d.", *grayPtr);
+						else
+							debugN(1, "....");
+					}
+#endif
+					if (*grayPtr > grayLevel)
+						*dst = 1;
+					else
+						*dst = 0;
+#if DRAWDEBUG
+					if (i == ccc) {
+						debugN("%c", *dst==1 ? '@' : '_');
+					}
+#endif					
+				}
+#if DRAWDEBUG
+				if(i == ccc)
+					debugN("\n");
+#endif	
+			}
+			const int bytes = dstPitch*h;
```

Comment:
This does not conform to our code formatting conventions.

Reply:
I will fix that.